src/service/user-service.ts

```diff
@@ -184,6 +189,35 @@ export default class UserService {
     return Promise.resolve(this.getSingleUser(user.id));
   }
 
+  /**
+   * Closes a user account by setting the user's status to deleted and inactive.
+   * Also, it sets the user's acceptedToS status to NOT_ACCEPTED and canGoIntoDebt to false.
+   * After saving these changes, it sends an account closure notification email to the user.
+   *
+   * @param {number} userId - The ID of the user to close the account for.
+   * @returns {Promise<void>} - A promise that resolves when the user account has been closed and the email has been sent.
+   * @throws {Error} - Throws an error if the email could not be sent.
+   */
+  public static async closeUser(userId: number): Promise<UserResponse> {
+    const user = await User.findOne({ where: { id: userId, deleted: false } });
+    if (!user) return undefined;
+    const currentBalance = await BalanceService.getBalance(user.id);
+
+    user.deleted = true;
+    user.active = false;
+    user.acceptedToS = TermsOfServiceStatus.NOT_ACCEPTED;
+    user.canGoIntoDebt = false;
+
+    await user.save().then(() => {
+      Mailer.getInstance().send(user, new AccountClosureNotification({
```

Comment:
Gewoon `user.save()` awaiten en dan deze mail sturen buiten een `.then()` blok. Die `await` bestaat namelijk exact voor deze use case.

Reply:
Het idee van de .then is dat je alleen een mail stuurt als het opslaan lukt, indien het opslaan niet werkt (en het account dus niet geclosed is) hoef je ook geen email te sturen. 

Je await ook de mail niet, deze zal dan ooit gestuurd worden.